### PR TITLE
feat(Notification): add experimental notification

### PR DESCRIPTION
**Pull request by emyarod**
TODO:

* [x] fix failing test due to a11y warning icon workaround

---

Closes IBM/carbon-components-react#1532

This PR will add support for the experimental `<InlineNotification />` and `<ToastNotification />` components

#### Changelog

**New**

* add support for experimental `<Notification />` component

**Changed**

* fixed notification icons not appearing for `<ToastNotification />`
* conditionally render icons based on notification type (based on new notification guidelines)

---

An open question: is there a reason for separating the inline and toast notification components rather than consolidating them in a common `<Notification />` component? It seems aside from `timeout`, `caption`, and minor markup changes, the majority of the code is shared between the two types of notifications. Would it be worth deduplicating this code?

### CODE REVIEW

File: src/components/Notification/Notification.js [hunk 1]
```diff
@@ -42,20 +42,17 @@ export class NotificationButton extends Component {
...
       {
         'bx--toast-notification__close-button': notificationType === 'toast',
         'bx--inline-notification__close-button': notificationType === 'inline',
       },
       className
     );
-
     const iconClasses = classNames({
-      'bx--toast-notification__icon': notificationType === 'toast',
+      'bx--toast-notification-icon': notificationType === 'toast',
```
Comment (asudoh) [inline src/components/Notification/Notification.js]: Looks that this change drifts away from https://github.com/IBM/carbon-components/blob/v9.46.0/src/components/notification/_toast-notification.scss#L139?

File: src/components/Notification/Notification.js [hunk 2]
```diff
@@ -42,20 +42,17 @@ export class NotificationButton extends Component {
       notificationType,
       ...other
     } = this.props;
-
```
Comment (asudoh) [inline src/components/Notification/Notification.js]: Nitpick, but wondering if a tool against blank lines is in use - We tend to keep blank lines wherever it's meaningful.

File: src/components/Notification/Notification.js [hunk 3]
```diff
@@ -158,16 +151,14 @@ export class ToastNotification extends Component {
   useIcon = kindProp =>
     ({
       error: iconErrorSolid,
-      info: iconInfoSolid,
       success: iconCheckmarkSolid,
-      warning: iconWarningSolid,
+      warning: iconErrorSolid,
```
Comment (asudoh) [inline src/components/Notification/Notification.js]: Just wanted to double-check if the icon list change here is intentional.

File: src/components/Notification/Notification.js [hunk 4]
```diff
@@ -181,15 +172,64 @@ export class ToastNotification extends Component {
...
       ...other
     } = this.props;
-
     const classes = classNames(
       'bx--toast-notification',
       { [`bx--toast-notification--${this.props.kind}`]: this.props.kind },
       className
     );
+    const a11yWarningIcon = (
+      <svg
```
Comment (asudoh) [inline src/components/Notification/Notification.js]: Probably we'd make it a separate React component in a separate module?

File: src/components/StructuredList/StructuredList-story.js [hunk 5]
```diff
@@ -12,6 +11,7 @@ import {
   StructuredListCell,
 } from '../StructuredList';
 import StructuredListSkeleton from '../StructuredList/StructuredList.Skeleton';
+import { componentsX } from '../../internal/FeatureFlags';
```
Comment (asudoh) [inline src/components/StructuredList/StructuredList-story.js]: Did you mean to put changes to this file in another PR? Same for `Tag.js`.

File: src/components/Notification/Notification.js (hunk 2 above)
Comment (emyarod) [reply src/components/Notification/Notification.js]: I can add them back, I was just trying to fit more lines on my screen for easier reading. This branch is still WIP so it just contained work I wanted to temporarily save and check into git

File: src/components/Notification/Notification.js (hunk 3 above)
Comment (emyarod) [reply src/components/Notification/Notification.js]: Thanks for catching that, my mistake! This branch is still WIP so it just contained work I wanted to temporarily save and check into git

File: src/components/StructuredList/StructuredList-story.js (hunk 5 above)
Comment (emyarod) [reply src/components/StructuredList/StructuredList-story.js]: since I created this branch off of those feature branches (tag and structured list) they contain the commits from those branches as well. I imagined the branches would be merged "in order" so I didn't split off from `master` branch each time. If it would be better for me to recreate these branches from `master` I can do that

File: src/components/Notification/Notification.js (hunk 1 above)
Comment (emyarod) [reply src/components/Notification/Notification.js]: I had some questions about that:

if I'm not mistaken, it looks like the component is using `bx--toast-notification-icon` here https://github.com/IBM/carbon-components/blob/v9.46.0/src/components/notification/notification.hbs#L88

do we want to sync the classes and styles on this component for the two libraries?

vanilla notification
image

React `bx--toast-notification-icon`
image

React `bx--toast-notification__icon`
image

based off of the `bx--inline-notification__close-icon` class, is it supposed to be `bx--toast-notification__close-icon` in both repos?

Comment (emyarod) [reply src/components/Notification/Notification.js]: going to go with `bx--toast-notification__close-icon` to follow @tw15egan's latest changes here https://github.com/IBM/carbon-components/pull/1412

File: src/components/Notification/Notification.js (hunk 4 above)
Comment (emyarod) [reply src/components/Notification/Notification.js]: sure, do we want to keep it within the `src/components/Notification` directory then?

**Comment by emyarod** [comment]
@asudoh just want to draw some discussion about the open question in the PR body:

> is there a reason for separating the inline and toast notification components rather than consolidating them in a common <Notification /> component? It seems aside from timeout, caption, and minor markup changes, the majority of the code is shared between the two types of notifications. Would it be worth deduplicating this code?

**Comment by emyarod** [comment]
component now ready for design review [here](http://react-experimental.carbondesignsystem.com/?selectedKind=Notifications)

**Comment by alisonjoseph** [comment]
I went ahead and merged this @emyarod. I believe there are some design tweaks needed for this but will be related to both vanilla/react. I think as long as our html matches then we are good to go.
